src/test/java/unit/CloneWorkspace.java

```diff
@@ -0,0 +1,200 @@
+package unit;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.hasSize;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import bio.terra.cli.serialization.userfacing.UFClonedResource;
+import bio.terra.cli.serialization.userfacing.UFClonedWorkspace;
+import bio.terra.cli.serialization.userfacing.UFResource;
+import bio.terra.cli.serialization.userfacing.UFWorkspace;
+import bio.terra.cli.serialization.userfacing.resource.UFBqDataset;
+import bio.terra.cli.serialization.userfacing.resource.UFGcsBucket;
+import bio.terra.workspace.model.CloneResourceResult;
+import bio.terra.workspace.model.StewardshipType;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.google.api.services.bigquery.model.DatasetReference;
+import harness.TestCommand;
+import harness.TestUsers;
+import harness.baseclasses.ClearContextUnit;
+import harness.utils.Auth;
+import harness.utils.ExternalBQDatasets;
+import java.io.IOException;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInfo;
+
+@Tag("unit")
+public class CloneWorkspace extends ClearContextUnit {
+  protected static final TestUsers workspaceCreator = TestUsers.chooseTestUserWithSpendAccess();
+  private DatasetReference externalDataset;
+  private UFWorkspace sourceWorkspace;
+  private UFWorkspace destinationWorkspace;
+
+  @BeforeEach
+  @Override
+  public void setupEachTime() throws IOException {
+    super.setupEachTime();
+  }
+
+  @AfterEach
+  public void cleanupEachTime() throws IOException {
+    if (sourceWorkspace != null) {
+      TestCommand.runCommandExpectSuccess(
```

Comment:
since this is for cleanup only, I think here and below should not expect success. if it was a single call to delete a workspace at the end of the cleanup method, that would be fine, but here if the source workspace delete fails, then it won't even attempt to delete the destination or external dataset. perhaps just run the command and log any failures.

also, you should login as the workspace creator in the cleanup method, in case the test methods didn't leave the auth in an appropriate state.

Reply:
good idea